internal/pkg/manifest/transform.go

```diff
@@ -18,58 +21,284 @@ var basicKinds = []reflect.Kind{
+			srcStruct := src.Convert(reflect.TypeOf(stringSliceOrString{})).Interface().(stringSliceOrString)
+			if err := mergo.Merge(&dstStruct, srcStruct, opts(basicTransformer{})...); err != nil {
+				return err
+			}
+			dst.Set(reflect.ValueOf(dstStruct).Convert(typ))
+			return nil
+		}
+		et.resetExclusiveFields = func(dst, src reflect.Value) error {
+			return resetExclusiveFields(dst, src, []string{"String"}, []string{"StringSlice"})
+		}
+	case reflect.TypeOf(PlatformArgsOrString{}):
+		et.merge = func(dst, src reflect.Value) error {
+			dstPlatformArgsOrString := dst.Interface().(PlatformArgsOrString)
+			srcPlatformArgsOrString := src.Interface().(PlatformArgsOrString)
+			if err := mergo.Merge(&dstPlatformArgsOrString, srcPlatformArgsOrString, opts(basicTransformer{})...); err != nil {
+				return err
+			}
+			dst.Set(reflect.ValueOf(dstPlatformArgsOrString))
+			return nil
+		}
+		et.resetExclusiveFields = func(dst, src reflect.Value) error {
+			return resetExclusiveFields(dst, src, []string{"PlatformString"}, []string{"PlatformArgs"})
+		}
+	case reflect.TypeOf(HealthCheckArgsOrString{}):
+		et.merge = func(dst, src reflect.Value) error {
+			dstHC := dst.Interface().(HealthCheckArgsOrString)
+			srcHC := src.Interface().(HealthCheckArgsOrString)
+			if err := mergo.Merge(&dstHC, srcHC, opts(basicTransformer{})...); err != nil {
+				return err
+			}
+			dst.Set(reflect.ValueOf(dstHC))
+			return nil
+		}
+		et.resetExclusiveFields = func(dst, src reflect.Value) error {
+			return resetExclusiveFields(dst, src, []string{"HealthCheckPath"}, []string{"HealthCheckArgs"})
+		}
+	case reflect.TypeOf(Count{}):
+		et.merge = func(dst, src reflect.Value) error {
+			dstC := dst.Interface().(Count)
+			srcC := src.Interface().(Count)
+			if err := mergo.Merge(&dstC, srcC, opts(countTransformer{})...); err != nil {
+				return err
+			}
+			dst.Set(reflect.ValueOf(dstC))
+			return nil
+		}
+		et.resetExclusiveFields = func(dst, src reflect.Value) error {
+			return resetExclusiveFields(dst, src, []string{"Value"}, []string{"AdvancedCount"})
+		}
+	default:
+		return nil // Return `nil` if this is not a composite type.
 	}
+	return et.transform
+}
 
-	if transform := transformBasic(typ); transform != nil {
+func (t workloadTransformer) transformExclusiveTypes(originalType reflect.Type) func(dst, src reflect.Value) error {
+	var et exclusiveTransformer
+	switch originalType {
+	case reflect.TypeOf(Image{}):
+		et.merge = func(dst, src reflect.Value) error {
+			dstImage := dst.Interface().(Image)
+			srcImage := src.Interface().(Image)
+			if err := mergo.Merge(&dstImage, srcImage, opts(imageTransformer{})...); err != nil {
+				return err
+			}
+			dst.Set(reflect.ValueOf(dstImage))
+			return nil
+		}
+		et.resetExclusiveFields = func(dst, src reflect.Value) error {
+			return resetExclusiveFields(dst, src, []string{"Build"}, []string{"Location"})
+		}
+	//case reflect.TypeOf(AdvancedCount{}):
```

Comment:
Any reason why we comment this one out? Is it because we haven't changed all the pointers of struct?

Reply:
No this code is not used anymore and I forgot to delete it. It's moved to count transformer.